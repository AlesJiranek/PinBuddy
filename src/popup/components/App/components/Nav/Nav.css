.nav {
  font: inheit;
  margin: 0;
  padding: 0;
  display: inline;
  height: var(--lh);
  line-height: var(--lh);
  color: var(--color-gray);
}

.nav:hover {
  color: var(--color-primary);
}

.nav--active {
  color: var(--color-primary);
}


.nav__spacer {
  margin: 0 calc(var(--lh) / 4);
}
