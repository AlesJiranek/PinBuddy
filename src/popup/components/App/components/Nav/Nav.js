import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { updateView, getInitialView } from 'redux-popup/actions/view';
import './Nav.css';

class Nav extends Component {
  componentDidMount() {
    this.props.getInitialView();
    window.addEventListener('keydown', this.handleKeydown);
  }

  componentWillUnmount() {
    window.removeEventListener('keydown', this.handleKeydown);
  }

  render() {
    const { username, online, view } = this.props;
    const shouldShowNav = username && online;

    return shouldShowNav && (
      <>
        <a
          href="all"
          data-location="all"
          title={`${chrome.i18n.getMessage('popupBrowseAllTitle')} ${navigator.platform === 'MacIntel' ? '(⌥ + 1)' : '(alt + 1)'}`}
          className={view === 'all' ? 'nav nav--active' : 'nav'}
          onClick={this.handleNavLinkClick}
        >
          {chrome.i18n.getMessage('popupBrowseAllText')}
        </a>

        <span className="nav__spacer">‧</span>

        <a
          href="add"
          data-location="add"
          title={`${chrome.i18n.getMessage('popupAddURLTitle')} ${navigator.platform === 'MacIntel' ? '(⌥ + 2)' : '(alt + 2)'}`}
          className={view === 'add' ? 'nav nav--active' : 'nav'}
          onClick={this.handleNavLinkClick}
        >
          {chrome.i18n.getMessage('popupAddURLText')}
        </a>
      </>
    );
  }

  handleNavLinkClick = e => {
    e.preventDefault();
    this.props.updateView(e.target.dataset.location);
  }

  handleKeydown = e => {
    //  ⌥ + 1
    if (e.keyCode === 49 && e.altKey) {
      e.preventDefault();
      this.props.updateView('all');
    }
    //  ⌥ + 2
    else if (e.keyCode === 50 && e.altKey) {
      e.preventDefault();
      this.props.updateView('add');
    }
  }
}

Nav.propTypes = {
  username: PropTypes.string.isRequired,
  online: PropTypes.bool.isRequired,
  view: PropTypes.string.isRequired,
  updateView: PropTypes.func.isRequired,
  getInitialView: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  username: state.user.username,
  online: state.online,
  view: state.view,
});

const mapDispatchToProps = dispatch => {
  return {
    updateView: location => dispatch(updateView(location)),
    getInitialView: () => dispatch(getInitialView()),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Nav);

