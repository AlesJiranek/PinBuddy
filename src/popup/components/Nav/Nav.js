import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { updateView } from '../../actions/view';
import './Nav.css';

class Nav extends Component {
  componentDidMount() {
    chrome.storage.sync.get([
      'defaultView',
    ], result => {
      if (result.defaultView) {
        this.props.updateView(result.defaultView);
      }
    });
  }
  render() {
    const { username, online, view } = this.props;

    return (username && online) ? (
      <>
        <a
          href="all"
          data-location="all"
          title={chrome.i18n.getMessage('popupBrowseAllTitle')}
          className={view === 'all' ? 'nav nav--active' : 'nav'}
          onClick={this.handleNavLinkClick}
        >
          {chrome.i18n.getMessage('popupBrowseAllText')}
        </a>

        <span className="nav__spacer">â€§</span>

        <a
          href="add"
          data-location="add"
          title={chrome.i18n.getMessage('popupAddURLTitle')}
          className={view === 'add' ? 'nav nav--active' : 'nav'}
          onClick={this.handleNavLinkClick}
        >
          {chrome.i18n.getMessage('popupAddURLText')}
        </a>
      </>
    ) : null;
  }

  handleNavLinkClick = e => {
    e.preventDefault();
    this.props.updateView(e.target.dataset.location);
  }
}

Nav.propTypes = {
  username: PropTypes.string.isRequired,
  view: PropTypes.string.isRequired,
  online: PropTypes.bool.isRequired,
  updateView: PropTypes.func.isRequired,
};

const mapStateToProps = (state, ownProps) => {
  return {
    username: state.user.username,
    online: state.online,
    view: state.view,
  };
};

const mapDispatchToProps = dispatch => {
  return {
    updateView: location => dispatch(updateView(location)),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Nav);

