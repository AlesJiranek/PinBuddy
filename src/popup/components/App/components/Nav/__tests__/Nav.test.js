import React from "react";
import { shallow } from "enzyme";

import { Nav } from "../Nav";

describe("<Nav />", () => {
  const defaultProps = {
    username: "pinbuddy",
    online: true,
    view: "all",
    error: false,
    updateView: jest.fn(),
    getInitialView: jest.fn()
  };

  it("should render correctly", () => {
    const tree = shallow(<Nav {...defaultProps} />);
    expect(tree).toMatchSnapshot();
  });
});
