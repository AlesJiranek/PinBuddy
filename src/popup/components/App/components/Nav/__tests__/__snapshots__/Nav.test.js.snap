// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav
    error={false}
    getInitialView={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    online={true}
    updateView={[MockFunction]}
    username="pinbuddy"
    view="all"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <a
          className="nav nav--active"
          data-location="all"
          href="all"
          onClick={[Function]}
          title="popupBrowseAllTitle (alt + 1)"
        >
          popupBrowseAllText
        </a>,
        <span
          className="nav__spacer"
        >
          ‧
        </span>,
        <a
          className="nav"
          data-location="add"
          href="add"
          onClick={[Function]}
          title="popupAddURLTitle (alt + 2)"
        >
          popupAddURLText
        </a>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "popupBrowseAllText",
          "className": "nav nav--active",
          "data-location": "all",
          "href": "all",
          "onClick": [Function],
          "title": "popupBrowseAllTitle (alt + 1)",
        },
        "ref": null,
        "rendered": "popupBrowseAllText",
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "‧",
          "className": "nav__spacer",
        },
        "ref": null,
        "rendered": "‧",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "popupAddURLText",
          "className": "nav",
          "data-location": "add",
          "href": "add",
          "onClick": [Function],
          "title": "popupAddURLTitle (alt + 2)",
        },
        "ref": null,
        "rendered": "popupAddURLText",
        "type": "a",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <a
            className="nav nav--active"
            data-location="all"
            href="all"
            onClick={[Function]}
            title="popupBrowseAllTitle (alt + 1)"
          >
            popupBrowseAllText
          </a>,
          <span
            className="nav__spacer"
          >
            ‧
          </span>,
          <a
            className="nav"
            data-location="add"
            href="add"
            onClick={[Function]}
            title="popupAddURLTitle (alt + 2)"
          >
            popupAddURLText
          </a>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "popupBrowseAllText",
            "className": "nav nav--active",
            "data-location": "all",
            "href": "all",
            "onClick": [Function],
            "title": "popupBrowseAllTitle (alt + 1)",
          },
          "ref": null,
          "rendered": "popupBrowseAllText",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "‧",
            "className": "nav__spacer",
          },
          "ref": null,
          "rendered": "‧",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "popupAddURLText",
            "className": "nav",
            "data-location": "add",
            "href": "add",
            "onClick": [Function],
            "title": "popupAddURLTitle (alt + 2)",
          },
          "ref": null,
          "rendered": "popupAddURLText",
          "type": "a",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
